File: apps/android-unity/Assets/Game/Core/Random/StableHash.cs
using System.Text;

namespace Baseball.Core.Random
{
    /// <summary>Stable FNV-1a 64-bit hashing over UTF-8 bytes.</summary>
    public static class StableHash
    {
        public static ulong Fnv1A64Value(string value)
        {
            unchecked
            {
                var hash = 0xCBF29CE484222325UL;
                var bytes = Encoding.UTF8.GetBytes(value ?? string.Empty);
                for (var index = 0; index < bytes.Length; index++)
                {
                    hash ^= bytes[index];
                    hash *= 0x00000100000001B3UL;
                }

                return hash;
            }
        }

        public static string Fnv1A64(string value) => Fnv1A64Value(value).ToString("x16");
    }
}

File: apps/android/game-core/src/main/kotlin/com/solkim/baseball/core/StableHash.kt
package com.solkim.baseball.core

import com.solkim.baseball.model.Hashing

/** Stable hash facade kept in the pure Kotlin core; values are part of the cross-runtime wire. */
public object StableHash {
    public fun fnv1a64(value: String): String = Hashing.fnv1a64Hex(value)

    public fun fnv1a64Value(value: String): ULong = Hashing.fnv1a64Hex(value).toULong(16)
}
